executor/aggfuncs/func_max_min.go

```diff
@@ -365,15 +599,44 @@ func (e *maxMin4Decimal) UpdatePartialResult(sctx sessionctx.Context, rowsInGrou
 		if isNull {
 			continue
 		}
-		if p.isNull {
-			p.val = *input
-			p.isNull = false
+		p.Push(*input)
```

Comment:
pushing a struct directly not a pointer,  it seems that the `map[interface{}]` can work Normal. I think i can use `toHashKey` to reduce the memory using of MyDecimal copy.